NeoMathEngine/src/CPU/x86/avx2/Avx2VectorFunctions.cpp

```diff
@@ -0,0 +1,270 @@
+/* Copyright © 2017-2023 ABBYY
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#include <NeoMathEngine/NeoMathEngineDefs.h>
+
+#ifdef NEOML_USE_SSE
+
+#include "Avx2Functions.h"
+
+#include <immintrin.h>
+#include <algorithm>
+
+static constexpr int AvxBlockSize = 8;
+
+#define AVX_LOAD_32_FLOATS(varPrefix, srcPtr) \
+	__m256 varPrefix##0 = _mm256_loadu_ps( srcPtr + 0 * AvxBlockSize ); \
+	__m256 varPrefix##1 = _mm256_loadu_ps( srcPtr + 1 * AvxBlockSize ); \
+	__m256 varPrefix##2 = _mm256_loadu_ps( srcPtr + 2 * AvxBlockSize ); \
+	__m256 varPrefix##3 = _mm256_loadu_ps( srcPtr + 3 * AvxBlockSize )
+
+#define AVX_STORE_32_FLOATS(varPrefix, dstPtr) \
+	_mm256_storeu_ps( dstPtr + 0 * AvxBlockSize, varPrefix##0 ); \
+	_mm256_storeu_ps( dstPtr + 1 * AvxBlockSize, varPrefix##1 ); \
+	_mm256_storeu_ps( dstPtr + 2 * AvxBlockSize, varPrefix##2 ); \
+	_mm256_storeu_ps( dstPtr + 3 * AvxBlockSize, varPrefix##3 )
+
+static const __m256i avxIOMask[AvxBlockSize] = {
```

Comment:
It's possible to define array:
`static const int avxIOMask[2 * AVXBlockSize - 1] = {0, 0, 0, 0, 0, 0, 0, 0, -1, -1, -1, -1, -1, -1, -1};`
and load to mask register from this array with different offset.